workspace: Add reporting for repository metadata

jwnimmer-tri opened the pull request:
In the future, we could use this for updates to mirrors and/or checking for new upstream releases.

jwnimmer-tri commented:
Sample output:
```
{
  "bullet": {
    "commit": "2.87",
    "name": "bullet",
    "repository": "bulletphysics/bullet3",
    "repository_rule_type": "github",
    "sha256": "438c151c48840fe3f902ec260d9496f8beb26dba4b17769a4a53212903935f95"
  },
  "ccd": {
    "commit": "63d3a911f016465a2ecf169d0c8bff8b601f1715",
    "name": "ccd",
    "repository": "danfis/libccd",
    "repository_rule_type": "github",
    "sha256": "1032cae04202330c5bcc9a652d75bef64669656bf4ea6cab74c5ff11c3f7a301"
  },
  "clang_cindex_python3": {
    "commit": "6a00cbc4a9b8e68b71caf7f774b3f9c753ae84d5",
    "name": "clang_cindex_python3",
    "repository": "wjakob/clang-cindex-python3",
    "repository_rule_type": "github",
    "sha256": "828e0d6238e2129a9e08071750dc16ba10e38eacf96f21b8a71e501c2085b282"
  },
  "fcl": {
    "commit": "3f59638077de81097336b300959dedcb60538d99",
    "name": "fcl",
    "repository": "flexible-collision-library/fcl",
    "repository_rule_type": "github",
    "sha256": "5b2d017920c5563b739b381a7721d375cfb6463a079af7fe5ccd8dbb3aeafff7"
  },
  "fmt": {
    "commit": "5.2.1",
    "name": "fmt",
    "repository": "fmtlib/fmt",
    "repository_rule_type": "github",
    "sha256": "3c812a18e9f72a88631ab4732a97ce9ef5bcbefb3235e9fd465f059ba204359b"
  },
  "godotengine": {
    "commit": "9bd402698cfa299b79b9144f8d7744c308a4e085",
    "name": "godotengine",
    "repository": "godotengine/godot",
    "repository_rule_type": "github",
    "sha256": "cbec9b48d79b1fdc16a253f438fb06544a59baf71cf2c1f47c193817f2127a10"
  },
  "gtest": {
    "commit": "release-1.8.1",
    "name": "gtest",
    "repository": "google/googletest",
    "repository_rule_type": "github",
    "sha256": "9bf1fe5182a604b4135edc1a425ae356c9ad15e9b23f9f12a02e80184c3a249c"
  },
  "json": {
    "commit": "v3.4.0",
    "name": "json",
    "repository": "nlohmann/json",
    "repository_rule_type": "github",
    "sha256": "c377963a95989270c943d522bfefe7b889ef5ed0e1e15d535fd6f6f16ed70732"
  },
  "lcm": {
    "commit": "v1.4.0",
    "name": "lcm",
    "repository": "lcm-proj/lcm",
    "repository_rule_type": "github",
    "sha256": "149d7076369d261e6adbb25d713dc9e30aeba415f4fc62bb41e748b2eb229b46"
  },
  "lcmtypes_bot2_core": {
    "commit": "9974c813bf746851067bb7b9adf86816c5039987",
    "name": "lcmtypes_bot2_core",
    "repository": "openhumanoids/bot_core_lcmtypes",
    "repository_rule_type": "github",
    "sha256": "a68d929f9e90b4c9433b4c18cd9eebcabe03c8082639cae5b3d6838a53f0001d"
  },
  "lcmtypes_robotlocomotion": {
    "commit": "821ff4b463a9cb8f616914d87289714fcb356a92",
    "name": "lcmtypes_robotlocomotion",
    "repository": "RobotLocomotion/lcmtypes",
    "repository_rule_type": "github",
    "sha256": "1407f8197950a6798908bd26c2e6186cf781f7e13ce91510da2b4823f4564953"
  },
  "meshcat": {
    "commit": "c7dcbca3af0e76f3c22573076a8d8e54b52dff67",
    "name": "meshcat",
    "repository": "rdeits/meshcat",
    "repository_rule_type": "github",
    "sha256": "4f2cd2e6cac98e1ffb8b60f13d2a8b15638bf38340e4f8f02d5101cb2babe226"
  },
  "meshcat_python": {
    "commit": "d0c8b6a9d1d750495ef9513254761cc14773cf99",
    "name": "meshcat_python",
    "repository": "rdeits/meshcat-python",
    "repository_rule_type": "github",
    "sha256": "99cdea957adf585b33c83cb284b6df16f953b0422ac5a035f4e0f50cf9105121"
  },
  "octomap": {
    "commit": "v1.9.0",
    "name": "octomap",
    "repository": "OctoMap/octomap",
    "repository_rule_type": "github",
    "sha256": "5f81c9a8cbc9526b2e725251cd3a829e5222a28201b394314002146d8b9214dd"
  },
  "optitrack_driver": {
    "commit": "5a015d5d7ae9e03c64359224ae0df2e68506bcef",
    "name": "optitrack_driver",
    "repository": "RobotLocomotion/optitrack-driver",
    "repository_rule_type": "github",
    "sha256": "52457d6f33aace87b00eaf2b2046119389a3e0f5b49a4a59b335b2f4a2e85c14"
  },
  "osqp": {
    "commit": "v0.4.1",
    "name": "osqp",
    "repository": "oxfordcontrol/osqp",
    "repository_rule_type": "github",
    "sha256": "0429d9580a9cdcce7c4ca61a9a97ccf4f24c7cab64f8dc44b3b9df73042cbac0"
  },
  "pybind11": {
    "commit": "5501496d6984d81bef3a0dacd9a2e368d544bfce",
    "name": "pybind11",
    "repository": "RobotLocomotion/pybind11",
    "repository_rule_type": "github",
    "sha256": "03f880238bdd4c29536448eb3999a75d38ff677ed1b3e38cdca0e8fd35753262"
  },
  "pycodestyle": {
    "commit": "2.3.1",
    "name": "pycodestyle",
    "repository": "PyCQA/pycodestyle",
    "repository_rule_type": "github",
    "sha256": "e9fc1ca3fd85648f45c0d2e33591b608a17d8b9b78e22c5f898e831351bacb03"
  },
  "pycps": {
    "commit": "83326ffaf03528e83f673eccb8af9b129504bf42",
    "name": "pycps",
    "repository": "mwoehlke/pycps",
    "repository_rule_type": "github",
    "sha256": "eede05c985d35f683445d8267be1ad7679b296b2d00290870c347e91822f51d1"
  },
  "qdldl": {
    "commit": "v0.1.3",
    "name": "qdldl",
    "repository": "oxfordcontrol/qdldl",
    "repository_rule_type": "github",
    "sha256": "a2c3a7d0c6a48b2fab7400fa8ca72a34fb1e3a19964b281c73564178f97afe54"
  },
  "scs": {
    "commit": "v2.0.2",
    "name": "scs",
    "repository": "cvxgrp/scs",
    "repository_rule_type": "github",
    "sha256": "8725291dfe952a1f117f1f725906843db392fe8d29eebd8feb14b49f25fc669e"
  },
  "spdlog": {
    "commit": "v1.2.1",
    "name": "spdlog",
    "repository": "gabime/spdlog",
    "repository_rule_type": "github",
    "sha256": "867a4b7cedf9805e6f76d3ca41889679054f7e5a3b67722fe6d0eae41852a767"
  },
  "stx": {
    "commit": "4f82af87fd97969a89ed0018ae72e5dacc8d308a",
    "name": "stx",
    "repository": "tcbrindle/cpp17_headers",
    "repository_rule_type": "github",
    "sha256": "7544aa3af121be49672e7dd227a4753bf063a10abda5c4cab9c094d423b6635e"
  },
  "styleguide": {
    "commit": "a61ad696eabc64ec06eb965b4925fdff7e433a36",
    "name": "styleguide",
    "repository": "RobotLocomotion/styleguide",
    "repository_rule_type": "github",
    "sha256": "18e5ddd60b54fd7181bcf3c917bc6fd7166faed097e71ca16463e87d5e7b7cc4"
  },
  "tinydir": {
    "commit": "1.2.4",
    "name": "tinydir",
    "repository": "cxong/tinydir",
    "repository_rule_type": "github",
    "sha256": "9c50eda69ba4854bb76ffa18961f49fd75f323b4cbebdf6b4b2d2db28f9f5ce2"
  },
  "tinyobjloader": {
    "commit": "v1.0.6",
    "name": "tinyobjloader",
    "repository": "syoyo/tinyobjloader",
    "repository_rule_type": "github",
    "sha256": "19ee82cd201761954dd833de551edb570e33b320d6027e0d91455faf7cd4c341"
  },
  "u_msgpack_python": {
    "commit": "v2.1",
    "name": "u_msgpack_python",
    "repository": "vsergeev/u-msgpack-python",
    "repository_rule_type": "github",
    "sha256": "38e780b2ecfd6bae09c1f82d6adc4ade2579558296cfa944751081fc5c7c6b29"
  }
}
```

jwnimmer-tri commented:
+@jamiesnape and +@EricCousineau-TRI for all reviews, please.  It's probably smart to do a quick first-pass design review before digging into details.  If you like the approach, then I'll also refactor `bitbucket.bzl` to have the same changes (in this PR).

jamiesnape commented:
It would make my life mildly easier if we printed the URL of the package, but I can live with `repository` and `repository_rule_type`, if necessary.

jamiesnape commented:
Adding `"url": "https://github.com/fmtlib/fmt/archive/5.2.1.tar.gz"` would be sufficient.

jwnimmer-tri commented:
I've updated the bitbucket rule now, and added urls to both bitbucket and github.  A revised snippet:
```
...
  "eigen": {
    "commit": "3.3.3",
    "name": "eigen",
    "repository": "eigen/eigen",
    "repository_rule_type": "bitbucket",
    "sha256": "94878cbfa27b0d0fbc64c00d4aafa137f678d5315ae62ba4aecddbd4269ae75f",
    "strip_prefix": "eigen-eigen-67e894c6cd8f",
    "urls": [
      "https://bitbucket.org/eigen/eigen/get/3.3.3.tar.gz",
      "https://drake-mirror.csail.mit.edu/bitbucket/eigen/eigen/3.3.3.tar.gz",
      "https://s3.amazonaws.com/drake-mirror/bitbucket/eigen/eigen/3.3.3.tar.gz"
    ]
  },
  "fcl": {
    "commit": "3f59638077de81097336b300959dedcb60538d99",
    "name": "fcl",
    "repository": "flexible-collision-library/fcl",
    "repository_rule_type": "github",
    "sha256": "5b2d017920c5563b739b381a7721d375cfb6463a079af7fe5ccd8dbb3aeafff7",
    "urls": [
      "https://github.com/flexible-collision-library/fcl/archive/3f59638077de81097336b300959dedcb60538d99.tar.gz",
      "https://drake-mirror.csail.mit.edu/github/flexible-collision-library/fcl/3f59638077de81097336b300959dedcb60538d99.tar.gz",
      "https://s3.amazonaws.com/drake-mirror/github/flexible-collision-library/fcl/3f59638077de81097336b300959dedcb60538d99.tar.gz"
    ]
  },
...
```

There is a lot of copypasta here, now.  Possibly I should try to refactor GH and BB to share more code.

jamiesnape commented:
Looks good.

jwnimmer-tri commented:
+@SeanCurtis-TRI for platform review per schedule, please.
